src/rust/engine/fs/src/store.rs

```diff
@@ -2033,6 +2109,48 @@ mod remote {
       }
       request
     }
+
+    //
+    /// Given a pre-created vector of digest(s), batch download then from Remote Store
+    ///
+    pub fn batch_download(
+      &self,
+      digests_batch: &Vec<Digest>,
+    ) -> impl future::Future<Item = HashSet<Digest>, Error = String> {
```

Comment:
Seems strange that this would return effectively the same thing as its argument; maybe it should return `()` rather than a `HashSet<Digest>`?